src/topology/algebra/module/linear_pmap.lean

```diff
@@ -0,0 +1,161 @@
+
+/-- A closed operator is trivially closable. -/
+lemma closed.closable {f : linear_pmap R E F} (hf : f.closed) : f.closable :=
+⟨f, hf.submodule_topological_closure_eq⟩
+
+/-- If `f` has a closable extension `g`, then `f` itself is closable. -/
+lemma closable.le_closable {f g : linear_pmap R E F} (hf : f.closable) (hfg : g ≤ f) : g.closable :=
+begin
+  cases hf with f' hf,
+  have : g.graph.topological_closure ≤ f'.graph :=
+  by { rw ←hf, exact submodule.topological_closure_mono (le_graph_of_le hfg) },
+  refine ⟨g.graph.topological_closure.to_linear_pmap _, _⟩,
+  { intros x hx hx',
+    cases x,
+    exact f'.graph_fst_eq_zero_snd (this hx) hx' },
+  rw [submodule.to_linear_pmap_graph_eq],
+end
+
+/-- The closure is unique. -/
+lemma closable.exists_unique {f : linear_pmap R E F} (hf : f.closable) :
+  ∃! (f' : linear_pmap R E F), f.graph.topological_closure = f'.graph :=
+begin
+  refine exists_unique_of_exists_of_unique hf (λ _ _ hy₁ hy₂, eq_of_eq_graph _),
+  rw [←hy₁, ←hy₂],
+end
+
+/-- The closure of a closable operator. -/
+noncomputable
+def closable.closure {f : linear_pmap R E F} (hf : f.closable) : linear_pmap R E F :=
+hf.some
+
+/-- The closure (as a submodule) of the graph is equal to the graph of the closure
+  (as a `linear_pmap`). -/
+lemma closable.graph_closure_eq_closure_graph {f : linear_pmap R E F} (hf : f.closable) :
+  f.graph.topological_closure = hf.closure.graph :=
+hf.some_spec
+
+/-- A closable `linear_pmap` is contained in its closure. -/
+lemma closable.le_closure {f : linear_pmap R E F} (hf : f.closable) : f ≤ hf.closure :=
+begin
+  refine le_of_le_graph _,
+  rw ←hf.graph_closure_eq_closure_graph,
+  exact (graph f).submodule_topological_closure,
+end
+
+lemma closable.closure_mono {f g : linear_pmap R E F} (hf : f.closable) (hg : g.closable)
+  (h : f ≤ g) :
+  hf.closure ≤ hg.closure :=
+begin
+  refine le_of_le_graph _,
+  rw ←hf.graph_closure_eq_closure_graph,
+  rw ←hg.graph_closure_eq_closure_graph,
+  exact submodule.topological_closure_mono (le_graph_of_le h),
+end
+
+/-- The closure is closed. -/
+lemma closable.closure_closed {f : linear_pmap R E F} (hf : f.closable) : hf.closure.closed :=
+begin
+  rw [closed, ←hf.graph_closure_eq_closure_graph],
+  exact f.graph.is_closed_topological_closure,
+end
+
+lemma closable_iff_exists_closed_extension {f : linear_pmap R E F} : f.closable ↔
+  ∃ (g : linear_pmap R E F) (hg : g.closed), f ≤ g :=
+⟨λ h, ⟨h.closure, h.closure_closed, h.le_closure⟩, λ ⟨_, hg, h⟩, hg.closable.le_closable h⟩
+
+lemma congr_closure {f g : linear_pmap R E F} (hf : f.closable) (hg : g.closable) (h : f = g)
+  : hf.closure = hg.closure :=
+begin
+  refine eq_of_eq_graph _,
+  rw [←hf.graph_closure_eq_closure_graph, ←hg.graph_closure_eq_closure_graph, h],
+end
+
+/-! ### The core of a linear operator -/
+
+/-- A submodule `S` is a core of `f` if the closure of the restriction of `f` to `S` is again `f`.-/
+def core {f : linear_pmap R E F} {S : submodule R E} (hS : S ≤ f.domain)
```

Comment:
As the core is not unique, I would prefer `is_core`. Also, I would move `hS` to the definition, i.e., `is_core f S` would mean that `S` is included in the domain of `f`, and the closure is `f`. Also maybe move `hf` to the definition as well.

Reply:
I'd really have `has_core f S` instead of `has_core f hS` here.

Reply:
Fixed it, sorry. Was overlooking that point.